doc/compute-job-configuration.md

```diff
@@ -0,0 +1,390 @@
+```
+
+> **Note: The importance of the default states**
+>
+> Since compute engine gives the ability to define completely custom states based on your monitoring infrastructure output we must also tag some custom states with specific meaning. These states might not be present in the monitoring messages but are produced during computations by the compute engine according to a specific logic. So we need to "tie" some of the custom status we declare to a specific default state of service.
+
+
+- `"default_down": "DOWNTIME"`
+
+   means that whenever compute engine needs to produce a status for a scheduled downtime will mark it using the "DOWNTIME" state.
+
+- `"default_missing": "MISSING"`
+
+	means whenever compute engine decides that a service status must declared missing (because there is no information provided from the metric data) will mark it using the "MISSING" state.
+
+- `"default_unknown: "UNKNOWN"`
+
+	means whenever compute engine decides that must produce a service status to be considered unknown (for e.g. during recomputation requests) will mark it using the "UNKNOWN" state.
+
+The available operations are declared in the operations list using truth tables as follows:
+
+```json
+"operations": {
+	"AND":[],
+	"OR":[]
+}
+```
+
+Each operation consists of a JSON array used to describe a _truth table_. An example of such a _truth table_ is presented below:
+
+```json
+"operations": {
+	"AND": [
+	  { "A":"OK",       "B":"OK",       "X":"OK"       },
+	  { "A":"OK",       "B":"WARNING",  "X":"WARNING"  },
+	  { "A":"OK",       "B":"UNKNOWN",  "X":"UNKNOWN"  },
+	]
+}
+```
+
+Each element of the JSON array describes a row of the _truth table_ for example:
+
+```json
+{ "A":"OK", "B":"WARNING", "X":"WARNING"}
+```
+
+declares that in an algorithmic AND operation between two status states of *OK* and *WARNING* the result is *WARNING*
+
+In the ops file the user is able to declare any number of available monitoring states and any number of available custom operations on those states. The ARGO Compute Engine uses this information to create the corresponding _truth tables_ in memory.
+
+#### {TENANT_NAME}_{REPORT_ID}_cfg.json (per report)
+
+A _Report Configuration_ file is a JSON file that contains specific information needed during the computations such as grouping parameters, the name of the availability profile used and many more. If the tenant's name is `T1` and the report name is `ReportA` then the filename of the config file must be `T1_ReportA_cfg.json`
+
+The configuration file of the job contains mandatory and optional fields with rich information describing the paramaters of the specific job. Some important fields are:
+
+```json
+"tenant": "tenant_name"`
+"job": "job_name",
+"aprofile": "availability_profile_name",
+"egroup": "endpoint_group_type_name",
+"ggroup": "group_of_group_type_name",
+"weight": "weight_factor_type_name"
+```
+[](Comment: Are the egroup and ggroup correct? Needs review by @kaggis)
```

Comment:
At the moment yes they are correct. This is how they are declared in the cfg files of the CE 

this is going to change because they are expressed more correctly in the new reports definitions in the API.